packages/thrift-client/src/main/connections/request.ts

```diff
@@ -32,11 +29,14 @@ export class RequestConnection<TClient> extends HttpConnection<TClient> {
         .post({
           body: dataToWrite,
           headers: {
-            'Content-length': dataToWrite.length,
+            'content-length': dataToWrite.length,
+            'content-type': 'application/octet-stream',
           },
         }, (err: any, response: request.RequestResponse, body: Buffer) => {
           if (err !== null) {
             reject(err)
+          } else if (response.statusCode !== 200) {
```

Comment:
Is there an isSuccess() function? Many HTTP libraries have one. There are other responses besides 200 that may mean success. I don't know if they apply here.

Reply:
For the thrift client we should always expect a 200. 2xx responses in general shouldn't apply.

Reply:
I've definitely seen 204 for some services (outside of CK). A check for 2xx may be safer depending on the target service.

Reply:
If you've seen it outside of CK I should handle it here since this is open source.